tests/ci/sqltest.py

```diff
@@ -0,0 +1,156 @@
+from env_helper import (
+    GITHUB_RUN_URL,
+    REPORTS_PATH,
+    TEMP_PATH,
+)
+from get_robot_token import get_best_robot_token
+from pr_info import PRInfo
+from report import TestResult
+from s3_helper import S3Helper
+from stopwatch import Stopwatch
+
+IMAGE_NAME = "clickhouse/sqltest"
+
+
+def get_run_command(pr_number, sha, download_url, workspace_path, image):
+    return (
+        f"docker run "
+        # For sysctl
+        "--privileged "
+        "--network=host "
+        f"--volume={workspace_path}:/workspace "
+        "--cap-add syslog --cap-add sys_admin --cap-add=SYS_PTRACE "
+        f'-e PR_TO_TEST={pr_number} -e SHA_TO_TEST={sha} -e BINARY_URL_TO_DOWNLOAD="{download_url}" '
+        f"{image}"
+    )
+
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+
+    stopwatch = Stopwatch()
+
+    temp_path = TEMP_PATH
+    reports_path = REPORTS_PATH
+
+    check_name = sys.argv[1]
+
+    if not os.path.exists(temp_path):
+        os.makedirs(temp_path)
+
+    pr_info = PRInfo()
+
+    gh = Github(get_best_robot_token(), per_page=100)
+    commit = get_commit(gh, pr_info.sha)
+
+    rerun_helper = RerunHelper(commit, check_name)
+    if rerun_helper.is_already_finished_by_status():
+        logging.info("Check is already finished according to github status, exiting")
+        sys.exit(0)
+
+    docker_image = get_image_with_version(reports_path, IMAGE_NAME)
+
+    build_name = get_build_name_for_check(check_name)
+    print(build_name)
+    urls = read_build_urls(build_name, reports_path)
+    if not urls:
+        raise Exception("No build URLs found")
+
+    for url in urls:
+        if url.endswith("/clickhouse"):
+            build_url = url
+            break
+    else:
+        raise Exception("Cannot find the clickhouse binary among build results")
+
+    logging.info("Got build url %s", build_url)
+
+    workspace_path = os.path.join(temp_path, "workspace")
+    if not os.path.exists(workspace_path):
+        os.makedirs(workspace_path)
+
+    run_command = get_run_command(
+        pr_info.number, pr_info.sha, build_url, workspace_path, docker_image
+    )
+    logging.info("Going to run %s", run_command)
+
+    run_log_path = os.path.join(temp_path, "run.log")
+    with open(run_log_path, "w", encoding="utf-8") as log:
+        with subprocess.Popen(
+            run_command, shell=True, stderr=log, stdout=log
+        ) as process:
+            retcode = process.wait()
+            if retcode == 0:
+                logging.info("Run successfully")
+            else:
+                logging.info("Run failed")
+
+    subprocess.check_call(f"sudo chown -R ubuntu:ubuntu {temp_path}", shell=True)
+
+    check_name_lower = (
+        check_name.lower().replace("(", "").replace(")", "").replace(" ", "")
+    )
+    s3_prefix = f"{pr_info.number}/{pr_info.sha}/sqltest_{check_name_lower}/"
+    paths = {
+        "run.log": run_log_path,
+        "main.log": os.path.join(workspace_path, "main.log"),
```

Comment:
just a side note, py3 has `pathlib.Path` class. It's much nicer in usage than the old `os.path` things